### PR TITLE
Adds cacheEnabled, cookiesEnabled, schemes to WebView

### PR DESCRIPTION
**Pull request by lwansbrough**
Adds three new properties to WebViews:
- cacheEnabled: (Boolean) Specify whether or not page content should be cached
- cookiesEnabled: (Boolean) Specify whether or not cookies should be used
- schemes: (Object of shape: { String: Boolean, ... }) A black/white list of schemes that are (dis)allowed. True means allowed, false means disallowed. By default all schemes are allowed, and the wildcard `*` may be used to target all schemes.

### CODE REVIEW

File: React/Views/RCTWebView.m [hunk 1]
```diff
@@ -45,6 +45,7 @@ - (instancetype)initWithEventDispatcher:(RCTEventDispatcher *)eventDispatcher
     _automaticallyAdjustContentInsets = YES;
     _contentInset = UIEdgeInsetsZero;
     _eventDispatcher = eventDispatcher;
+    _schemes = [NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"*"];
```
Comment (ide) [inline React/Views/RCTWebView.m]: `@YES`

File: React/Views/RCTWebView.m [hunk 2]
```diff
@@ -158,9 +177,26 @@ - (BOOL)webView:(__unused UIWebView *)webView shouldStartLoadWithRequest:(NSURLR
     }];
     [_eventDispatcher sendInputEventWithName:@"topLoadingStart" body:event];
   }
+  
+  NSString *scheme = request.URL.scheme;
+  
+  NSNumber *catchAllScheme;
+  if ([_schemes objectForKey:@"*"] == nil) {
+    catchAllScheme = [NSNumber numberWithBool:YES];
```
Comment (ide) [inline React/Views/RCTWebView.m]: `@YES`

File: React/Views/RCTWebView.m [hunk 3]
```diff
@@ -158,9 +177,26 @@ - (BOOL)webView:(__unused UIWebView *)webView shouldStartLoadWithRequest:(NSURLR
...
+  }
+  else {
+    catchAllScheme = [_schemes valueForKey:@"*"];
```
Comment (ide) [inline React/Views/RCTWebView.m]: `_schemes[@"*"]` and elsewhere in this diff

File: React/Views/RCTWebView.m (hunk 1 above)
Comment (lwansbrough) [inline React/Views/RCTWebView.m]: @ide Can you explain the difference?

Comment (ide) [inline React/Views/RCTWebView.m]: @YES is the modern syntax.

Comment (lwansbrough) [inline React/Views/RCTWebView.m]: @ide Ah, so it's `@YES` instead of `[NSNumber numberWithBool:YES]`?

Comment (ide) [inline React/Views/RCTWebView.m]: `@YES` ;)

Comment (ide) [inline React/Views/RCTWebView.m]: Also in this specific case you could write `@{@"*": @YES}`

Comment (lwansbrough) [inline React/Views/RCTWebView.m]: Noted, thanks. Will commit these changes shortly.

**Comment by alexisvincent** [comment]
@lwansbrough Status on this?

**Comment by alexisvincent** [comment]
@lwansbrough How would i feasibly get web view caching working locally?

**Comment by lwansbrough** [comment]
@alexisvincent Caching is usually enabled by default. I added it as an option because I needed to disable it.

**Comment by alexisvincent** [comment]
@lwansbrough I have a situation where its not caching assets. However come caches them... Any ideas?

**Comment by facebook-github-bot-8** [comment]
@alexeylang would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
Who's the right person to look at this? @javache do you have any context on iOS WebViews?

**Comment by viesti** [comment]
Would need similar for Android too, maybe something like this:

```
  @ReactProp(name = "clearCache")
  public void clearCache(WebView view, boolean includeDiskFiles) {
    view.clearCache(includeDiskFiles);
  }
```

in ```react-native/ReactAndroid/src/main/java/com/facebook/react/views/webview/ReactWebViewManager.java```

**Comment by facebook-github-bot-3** [comment]
It's been a while since the last commit was reviewed and the labels show this pull request needs review. Based on the blame information for the files in this pull request we identified @nicklockwood as a potential reviewer. Could you take a look please or cc someone with more context?

**Comment by lacker** [comment]
As far as I can tell this pull request has been abandoned because there are some comments not responded to by the author. I'm going to close it, but if I'm mistaken and this is actually waiting on a code review then please feel free to reopen it.

**Comment by booboothefool** [comment]
Would appreciate this. Currently dealing with a case where my WebView is caching when I really don't want it to. :c

**Comment by aqnaruto** [comment]
great work, thank you!@lwansbrough

**Comment by aqnaruto** [comment]
@booboothefool no, it is not cached

**Pull request closed without merging**